### DISCUSSION TITLE
New Tuya Devices

### DISCUSSION THREAD

**Discussion by mutozorro** [make-all/tuya-local#1027] [Show and tell]
I have two yaml files that work with two my tuya devices. It would be great if you can add them into the device folder. Otherwise, I need to manually add them any time a new component revision is released. Thanks

**_EMylo WiFi Smart Meter_**

name: EMylo WiFi Smart Meter
products:
  - id: nqbs1onwskmmmaac
    name: YSI115D5
primary_entity:
  entity: sensor
  name: Energy
  class: energy
  dps:
    - id: 1
      type: integer
      name: sensor
      unit: kWh
      class: total_increasing
      mapping:
        - scale: 100
    - id: 6
      type: string
      name: phase_a
      optional: true
    - id: 10
      type: bitfield
      name: fault_code
    - id: 17
      type: string
      name: alarm
      optional: true
secondary_entities:
  - entity: binary_sensor
    name: Fault
    class: problem
    category: diagnostic
    dps:
      - id: 10
        type: bitfield
        name: sensor
        mapping:
          - dps_val: 0
            value: false
          - value: true
  - entity: switch
    name: Prepayment
    category: config
    icon: "mdi:cash"
    dps:
      - id: 12
        type: boolean
        name: switch
  - entity: sensor
    name: Prepaid energy
    category: diagnostic
    dps:
      - id: 13
        type: integer
        name: sensor
        unit: kWh
        optional: true
        mapping:
          - scale: 100
  - entity: button
    name: Clear energy
    class: reset
    category: config
    dps:
      - id: 14
        type: boolean
        name: button
        optional: true
  - entity: number
    name: Charge energy
    category: config
    icon: "mdi:cash"
    dps:
      - id: 15
        type: integer
        name: value
        unit: kWh
        optional: true
        range:
          min: 0
          max: 50000
  - entity: switch
    dps:
      - id: 16
        type: boolean
        name: switch
  - entity: button
    name: Clear event
    class: reset
    category: config
    dps:
      - id: 20
        type: boolean
        name: button
        optional: true

_**TOMZN WIFI Energy Meter**_

name: TOMZN WIFI Energy Meter
products:
  - id: rrmpzckqm5pfd642
    name: Breaker
primary_entity:
  entity: switch
  dps:
    - id: 16
      name: switch
      type: boolean
    - id: 17
      name: alarm_set_1
      optional: true
      type: string
    - id: 18
      name: alarm_set_2
      optional: true
      type: string
    - id: 6
      name: phase_a
      type: string
      optional: true
    - id: 9
      name: fault
      type: bitfield
      optional: true
      mapping:
        - dps_val: 0
          value: OK
    - id: 19
      name: breaker_number
      type: string
secondary_entities:
  - entity: switch
    name: Prepayment
    icon: "mdi:cash-multiple"
    category: config
    dps:
      - id: 11
        type: boolean
        name: switch
  - entity: button
    name: Energy reset
    class: restart
    category: config
    dps:
      - id: 12
        type: string
        name: button
        optional: true
        mapping:
          - dps_val: empty
            value: true
  - entity: sensor
    name: Balance energy
    category: diagnostic
    dps:
      - id: 13
        type: integer
        name: sensor
        unit: kWh
        mapping:
          - scale: 10
  - entity: number
    name: Charge energy
    category: config
    icon: "mdi:cash"
    dps:
      - id: 14
        type: integer
        name: value
        optional: true
        unit: kWh
        range:
          min: 0
          max: 99999
        mapping:
          - scale: 10
  - entity: sensor
    name: Leakage current
    class: current
    category: diagnostic
    dps:
      - id: 15
        type: integer
        name: sensor
        unit: mA
        class: measurement
  - entity: button
    name: Earth leak test
    icon: "mdi:current-ac"
    category: config
    dps:
      - id: 21
        type: boolean
        name: button
        optional: true
  - entity: sensor
    name: Energy
    class: energy
    dps:
      - id: 1
        type: integer
        name: sensor
        unit: kWh
        class: total_increasing
        mapping:
          - scale: 100

**Comment by make-all** [comment]
I'm confused. These two devices are both already supported.

**Comment by mutozorro** [comment]
Ciao Jason, when I try to configure my devices using only the original device’s list, for example after an upgrade to a new version, my devices don’t work. In order to get them working I need to add in the directory \config\custom_components\tuya_local\devices the two files that I posted on GitHub. I don’t know why… Regards  Inviato da Posta per Windows Da: Jason RumneyInviato: sabato 26 agosto 2023 11:18A: make-all/tuya-localCc: mutozorro; AuthorOggetto: Re: [make-all/tuya-local] New Tuya Devices (Discussion #1027) I'm confused. These two devices are both already supported.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by mutozorro** [comment]
Hi Jason,another tuya_local revision was released and again the same problem!
The integration was working fine under revision 2023.8.2


I updated to revision 2023.9.0 and this was the result



To have the integration work again I had to copy my configuration files into \config\custom_components\tuya_local\devicesand reload the integration.
Is it too much trouble to add these two files to the distribution package?
Thanks and Best Regards

    Il sabato 26 agosto 2023 alle ore 13:35:52 CEST, a c ***@***.***> ha scritto:  
 
 
Ciao Jason,

  

when I try to configure my devices using only the original device’s list, for example after an upgrade to a new version, my devices don’t work. In order to get them working I need to add in the directory \config\custom_components\tuya_local\devices the two files that I posted on GitHub. I don’t know why…

  

Regards

  

  

Inviato da Posta per Windows

  

Da: Jason Rumney
Inviato: sabato 26 agosto 2023 11:18
A: make-all/tuya-local
Cc: mutozorro; Author
Oggetto: Re: [make-all/tuya-local] New Tuya Devices (Discussion #1027)

  

I'm confused. These two devices are both already supported.

**Comment by Bt338** [comment]
Hi Jason, I have a yaml files that works with IH-K009 zigbee temperature and humidity sensors. Can you add it to the device folder? Thanks

name: Climate sensor
products:
  - id: dowj6gyi
    name: IH-K009
primary_entity:
  entity: sensor
  class: temperature
  dps:
    - id: 1
      type: integer
      name: sensor
      unit: C
      class: measurement
      mapping:
        - scale: 10
secondary_entities:
  - entity: sensor
    class: humidity
    dps:
      - id: 2
        type: integer
        name: sensor
        unit: "%"
        class: measurement
        mapping:
          - scale: 10
  - entity: sensor
    class: battery
    category: diagnostic
    dps:
      - id: 4
        type: integer
        name: sensor
        unit: "%"
        class: measurement
        mapping:
          - scale: 1

**Comment by make-all** [comment]
This matches the zigbee_zn373186_temphumid_sensor config
